tests/suites/test_suite_pk.function

```diff
@@ -594,8 +594,8 @@ void pk_rsa_alt(  )
     mbedtls_rsa_context raw;
     mbedtls_pk_context rsa, alt;
     mbedtls_pk_debug_item dbg_items[10];
-    unsigned char hash[50], sig[1000];
-    unsigned char msg[50], ciph[1000], test[1000];
+    unsigned char hash[50], sig[64];
+    unsigned char msg[50], ciph[64], test[50];
```

Comment:
How were these numbers determined?

Reply:
I ran the tests on PC and printed the values.

Reply:
How fragile are these values? How likely is it for these values to become incorrect?

Reply:
These are the lengths from the *current* test:
~~~
RSA encrypt test vector ........................................... test_pk_rsa_encrypt_test_vec output length is 256
PASS
RSA decrypt test vector #1 ........................................ test_pk_rsa_decrypt_test_vec output length is 16
PASS
RSA decrypt test vector #2 ........................................ test_pk_rsa_decrypt_test_vec output length is 0
PASS
EC nocrypt ........................................................ test_pk_ec_nocrypt output encrypt length is 0
test_pk_ec_nocrypt output decrypt length is 0
PASS
EC-DH nocrypt ..................................................... test_pk_ec_nocrypt output encrypt length is 0
test_pk_ec_nocrypt output decrypt length is 0
PASS
ECDSA nocrypt ..................................................... test_pk_ec_nocrypt output encrypt length is 0
test_pk_ec_nocrypt output decrypt length is 0
PASS
RSA_ALT consistency ............................................... test_pk_rsa_alt sig length is 64
test_pk_rsa_alt cipher length is 64
test_pk_rsa_alt test length is 50
test_pk_rsa_alt cipher2 length is 64
PASS
~~~
Note, that when I set the hash length to `MBEDTLS_MD_MAX_SIZE` (64 when SHA512 defined), I got a failure, because of:
In rsa_rsassa_pkcs1_v15_encode, when `md_alg` is `MBEDTLS_MD_NONE`:
~~~
       if( nb_pad < hashlen )
            return( MBEDTLS_ERR_RSA_BAD_INPUT_DATA );

and then 
 if( nb_pad < 3 + 8 )
        return( MBEDTLS_ERR_RSA_BAD_INPUT_DATA );
~~~
`nb_pad` is 64, so after reducing the `hashlen` which is 64, we fail because `0 < 3 + 8`

Reply:
AS you can see in the [`test_data`](https://github.com/ARMmbed/mbedtls/blob/development/tests/suites/test_suite_pk.data#L81), the test doesn't get any parameters, and the sizes are for internal use, so I wouldn't say the sizes in this specific test are fragile